optuna/trial/_trial.py

```diff
@@ -384,14 +388,26 @@ def objective(trial):
                 Lower endpoint of the range of suggested values. ``low`` is included in the range.
             high:
                 Upper endpoint of the range of suggested values. ``high`` is included in the range.
-            step:
-                A step of spacing between values.
-
-        Returns:
-            A suggested integer value.
+            log:
+                A flag to sample the value from the log domain or not.
+                If ``log`` is true,
+                the range of suggested values is converted to the log domain:
+                ``[log(low-0.5), log(high+0.5)]``,
+                and sample a float value ``u`` uniformly distributed in the range.
+                Then it is reversed and transform to the original domain:
+                ``round((exp(u) - low) / step) * step + low``.
+                For example,
+                if `low = 2`, `high = 8` and `step = 2`,
+                then the suggested value is in `[2, 4, 6, 8]`
```

Comment:
It might be good to clarify how the range of suggested values is divided.
```suggestion
                then the range of suggested values is divided by ``step`` as `[2, 4, 6, 8]`
```